packages/big-design/src/components/Worksheet/hooks/useEditableCell/useEditableCell.ts

```diff
@@ -40,9 +41,22 @@ export const useEditableCell = <T extends WorksheetItem>(cell: Cell<T>) => {
     }
   }, [cell, setEditingCell]);
 
-  const handleBlur = useCallback(() => {
-    restoreFocus();
-  }, [restoreFocus]);
+  const handleBlur = useCallback(
+    (event?: FocusEvent<HTMLInputElement>) => {
+      const isNumberCell = selectedCells[0].type === 'number';
+      const isTextCell = selectedCells[0].type === 'text';
+
+      if (isNumberCell || isTextCell) {
+        updateItems(
+          [selectedCells[0]],
+          [isNumberCell ? Number(event?.target.value) : event?.target.value],
+        );
+      }
+
+      restoreFocus();
+    },
+    [restoreFocus, selectedCells, updateItems],
+  );
```

Comment:
Hmm if we had a multiple selection of cells (in the future), how would this be managed? Can we pass the the `cell` as a param instead?

Reply:
@jorgemoya This behavior should work only for the first selected cell. What do you mean by passing the `cell` as a param?